### DISCUSSION TITLE
Support `#<<` to highlight lines in revealjs?

### DISCUSSION THREAD

**Discussion by etiennebacher** [quarto-dev/quarto-cli#2627] [Feature Requests]
I like the way to highlight lines in revealjs slides with the chunk option `code-line-numbers`, but one drawback to me is that the line numbers are hardcoded. This means that if I want to modify the chunk later, I have to change the line numbers, which is a bit annoying.

Before using `revealjs`, I used `xaringan` which supported the special comment `#<<` to highlight specific lines, like this:
````rmd
---
output: 
  xaringan::moon_reader:
    nature: 
      highlightLines: true
---

```{r}
# Higlight the following line
print("a") #<<
```
````
<img width="522" alt="foo" src="https://user-images.githubusercontent.com/52219252/192531424-b3c05d12-06e4-4ce5-9773-1781f08ca988.PNG">

Would it be possible to support something like this?

**Comment by mcanouil** [comment]
It's unlikely to happen because this is Pandoc/Reveal.js internal framework.
This would require to write a pre-processor script (which you could write if that is something you really need, see <https://quarto.org/docs/projects/scripts.html#pre-and-post-render>) that will read the code cells, detect the symbol `#<<`, figure out the line number, set the code cell option `code-line-numbers`.

> but one drawback to me is that the line numbers are hardcoded

So is your `#<<` at the end of the line.

**Reply by etiennebacher** [reply]
> >  but one drawback to me is that the line numbers are hardcoded
>
> So is your #<< at the end of the line.

Yes but if I add or remove lines above `print("a") #<<`, the line will still be highlighted. With hardcoded line numbers, if I add or remove lines above, I have to change the line number to highlight

**Reply by mcanouil** [reply]
I can see how it can be a little inconvenient, but for presentation, do you have to change that often your code and the highlighting?

I feel the issue might be more on the process to build your slides rather than on the tool to build them. 🤔

**Reply by etiennebacher** [reply]
For sure it's not the most important feature ever, it just makes it a bit more comfortable. And since it was something available in `xaringan` I thought it might be implemented in `revealjs` as well. But if it's hard to implement then it's ok.

And regarding the process of making slides, it's also due to the fact that I don't teach a lot so I probably don't have the most efficient workflow, but sometimes you're like "I should add a comment in this chunk of code" and if you have a lot of chunks then it's a bit annoying to change all line numbers to highlight.

**Reply by mcanouil** [reply]
Such feature can be implemented, but should it? (I am not the one that can answer that)
`knitr` hooks and/or pre-render script could be used but again is this time worth spending?
When I make slides, the highlighting, etc. is one of the final step and only once, so I never had the issue to change so often the highlighting/etc. that it became a hassle.

For your information, `xaringan` is built on top of Remark.js while Quarto is using Pandoc which uses Reveal.js.
Remark.js and Reveal.js are two totally different framework.

**Comment by mcanouil** [comment]
I believe the "code annotation" feature addresses this, see <https://quarto.org/docs/authoring/code-annotation.html> (if not feel free to reopen).
